src/integrationtests/java/com/aws/greengrass/integrationtests/logmanager/LogManagerConfigTest.java

```diff
@@ -0,0 +1,327 @@
+            URISyntaxException, IOException, DeviceConfigurationException {
+
+        tempRootDir = Paths.get(System.getProperty("root"));
+        tempDirectoryPath = Files.createTempDirectory(tempRootDir, "IntegrationTestsTemporaryLogFiles");
+
+        CountDownLatch logManagerRunning = new CountDownLatch(1);
+        kernel = new Kernel();
+
+        Path testRecipePath = Paths.get(LogManagerTest.class.getResource("configsDifferentFromDefaults.yaml").toURI());
+        String content = new String(Files.readAllBytes(testRecipePath), StandardCharsets.UTF_8);
+        content = content.replaceAll("\\{\\{logFileDirectoryPath}}",
+                tempDirectoryPath.toAbsolutePath().toString());
+
+        Map<String, Object> objectMap = YAML_OBJECT_MAPPER.readValue(content, Map.class);
+        kernel.parseArgs();
+        kernel.getConfig().mergeMap(System.currentTimeMillis(), ConfigPlatformResolver.resolvePlatformMap(objectMap));
+
+        kernel.getContext().addGlobalStateChangeListener((service, oldState, newState) -> {
+            if (service.getName().equals(LogManagerService.LOGS_UPLOADER_SERVICE_TOPICS)
+                    && newState.equals(State.RUNNING)) {
+                logManagerRunning.countDown();
+                logManagerService = (LogManagerService) service;
+            }
+        });
+        deviceConfiguration = new DeviceConfiguration(kernel, "ThingName", "xxxxxx-ats.iot.us-east-1.amazonaws.com", "xxxxxx.credentials.iot.us-east-1.amazonaws.com", "privKeyFilePath",
+                "certFilePath", "caFilePath", "us-east-1", "roleAliasName");
+
+        kernel.getContext().put(DeviceConfiguration.class, deviceConfiguration);
+        kernel.launch();
+        assertTrue(logManagerRunning.await(10, TimeUnit.SECONDS));
+    }
+
+    @AfterAll
+    static void afterAll() {
+        kernel.shutdown();
+    }
+
+    @Test
+    void GIVEN_periodicUploadIntervalSec_config_WHEN_value_is_reset_and_replaced_THEN_correct_values_are_used() {
+            assertThat(()-> logManagerService.getPeriodicUpdateIntervalSec(), eventuallyEval(is(60),
+                    Duration.ofSeconds(30)));
+
+            logManagerService.getConfig().find(CONFIGURATION_CONFIG_KEY, LOGS_UPLOADER_PERIODIC_UPDATE_INTERVAL_SEC).remove();
+            assertThat(()-> logManagerService.getPeriodicUpdateIntervalSec(),
+                    eventuallyEval(is(LogManagerService.DEFAULT_PERIODIC_UPDATE_INTERVAL_SEC), Duration.ofSeconds(30)));
+
+            logManagerService.getConfig().lookup(CONFIGURATION_CONFIG_KEY, LOGS_UPLOADER_PERIODIC_UPDATE_INTERVAL_SEC).withValue(600);
+            assertThat(()-> logManagerService.getPeriodicUpdateIntervalSec(), eventuallyEval(is(600),
+                    Duration.ofSeconds(30)));
+    }
+
+    @Test
+    void GIVEN_uploadToCloudWatch_config_WHEN_value_is_reset_and_replaced_THEN_correct_values_are_used() {
+        // uploadToCloudWatch=true in recipe, so system configs expected in componentLogConfigurations map
```

Comment:
I remember it's not in recipe. But the yaml config we used here has system configs. So I think this test case needs to assert it's true at first, then do removal and assert that it resets to null/false.

Reply:
Agreed that since it's true in the yaml file, the test needs to first verify that it's true, then remove it and verify that it's false, and then replace with true and verify that it's true again. For uploadToCloudWatch, the value itself isn't saved into componentLogConfigurations,  but rather if it is false, the whole system component configs aren't added to componentLogConfigurations. So to verify whether uploadToCloudWatch is true/false, the test is checking whether or not componentLogConfigurations contains the system component configs.